blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueuedBlueRun.java

```diff
@@ -0,0 +1,159 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import com.google.common.collect.ImmutableList;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.hal.Links;
+import io.jenkins.blueocean.rest.model.BlueActionProxy;
+import io.jenkins.blueocean.rest.model.BlueArtifactContainer;
+import io.jenkins.blueocean.rest.model.BlueChangeSetEntry;
+import io.jenkins.blueocean.rest.model.BluePipelineNodeContainer;
+import io.jenkins.blueocean.rest.model.BluePipelineStepContainer;
+import io.jenkins.blueocean.rest.model.BlueRun;
+import io.jenkins.blueocean.rest.model.Container;
+import org.kohsuke.stapler.export.Exported;
+
+import java.util.Collection;
+import java.util.Date;
+
+public class QueuedBlueRun extends BlueRun {
+
+    public static final String QUEUE_ID = "queueId";
+
+    private final BlueRunState runState;
+    private final BlueRunResult runResult;
+    private final QueueItemImpl item;
+    private final Link parent;
+
+    public QueuedBlueRun(BlueRunState runState, BlueRunResult runResult, QueueItemImpl item, Link parent) {
+        this.runState = runState;
+        this.runResult = runResult;
+        this.item = item;
+        this.parent = parent;
+    }
+
+    @Override
+    public String getOrganization() {
+        return item.getOrganization();
+    }
+
+    @Override
+    public String getId() {
```

Comment:
Its quasi run id, its not run yet. This can be confusing to client as expected run id, might be reused and is simply a guess. Better to call it out what it is, that is `getExpectedRunId()` and client should treat it as such. Otherwise I am pretty sure someone later on is going to re-write this whole thing again:)